File: backend/src/main/java/com/pawconnect/backend/user/dto/CurrentUserResponse.java
package com.pawconnect.backend.user.dto;

import com.pawconnect.backend.dog.dto.DogResponse;
import com.pawconnect.backend.user.model.UserGender;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;

@Data
public class CurrentUserResponse {
    private Long id;
    private String username;
    private String email;
    private String bio;
    private LocalDate birthdate;
    private UserGender gender;
    private double latitude;
    private double longitude;
    private Boolean locationVisible;
    private String profilePhotoUrl;
    private Set<String> languages;
    private List<DogResponse> dogs;
}

File: mobile/lib/src/models/current_user_response.dart
class CurrentUserResponse {
  final int id;
  final String username;
  final String email;
  final String? bio;
  final String? birthdate;
  final String? gender;
  final double latitude;
  final double longitude;
  final bool? locationVisible;
  final String? profilePhotoUrl;
  final List<String> languages;
  final List<dynamic> dogs;

  CurrentUserResponse.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        username = json['username'],
        email = json['email'],
        bio = json['bio'],
        birthdate = json['birthdate'],
        gender = json['gender'],
        latitude = (json['latitude'] as num).toDouble(),
        longitude = (json['longitude'] as num).toDouble(),
        locationVisible = json['locationVisible'],
        profilePhotoUrl = json['profilePhotoUrl'],
        languages = List<String>.from(json['languages'] ?? []),
        dogs = List<dynamic>.from(json['dogs'] ?? []);
}
